### ISSUE TITLE
use custom (upstream) repo

### ISSUE THREAD

**Issue by fommil** [vermiculus/magithub#66]
My workflow is to pull from an upstream team repo and push to my origin (confusingly called "upstream" in git nomenclature). Magit handles this perfectly fine using it's concept of `u` in config parameters.

However, this doesn't work with magithub at present because it will use the default (origin) repo.

Please please please support alternative repos, defaulting to whatever magit decides is the pull upstream.

I'm quite happy to give you a donation via bountysource for this ticket for a nice bottle of your favourite beverage if that is something you would accept! :smile:

**Comment by vermiculus** [comment]
What would you imagine the status buffer to look like?

**Comment by fommil** [comment]
great to see you've pushed something here! :smile:

I'd just expect it to look the same way as the images I've seen of magithub already on your README. I've not been able to use it yet at all because all my work follows this workflow.

I'll try out your changes.

**Comment by vermiculus** [comment]
Yes, but would you interleave issues from different repositories? Replace your own issues with theirs? List one repository's issues/pulls before moving on to a next? Etc.

**Comment by fommil** [comment]
nah, just listing everything from the upstream one is perfect.

btw, please collect a monetary thank you :smile: https://www.bountysource.com/issues/41922867-use-custom-upstream-repo

**Comment by vermiculus** [comment]
You're so kind – I'm flattered! But I can't legally accept it :) Please consider [donating to Magit instead](https://magit.vc/donate) ❤️

Setting up a basic proxy-repo should be simple enough; I'll see what I can get moved in today.

**Comment by fommil** [comment]
I already do donate to magit... the easiest way to redirect this bounty is for @tarsius to claim it.

**Comment by vermiculus** [comment]
If @tarsius can't accept it directly, I can certainly accept and redirect it.

The changes should now be usable; you should be able to set a proxy URL with `M-x magithub-proxy-set-default` (bound to `H x` in the status buffer). Once you do that, the issue/PR list should automatically pull in the real issues.

I'm so happy Magithub has made your life easier 😄

**Comment by fommil** [comment]
awesome, this is showing me PRs and issues!

Now if only there was a way to prioritise them :stuck_out_tongue:

**Comment by vermiculus** [comment]
That sounds like a new issue 😉 It might be possible. It would certainly be easier if GitHub supported this kind of thing, but `/.git/config` could be abused to support this kind of thing.

**Comment by tarsius** [comment]
Just wanted to say that I saw the reward forwarding thing above, but was a bit to busy to respond. It seems @vermiculus has decided by now that he can claim it after all. Good for him ;-)

**Comment by vermiculus** [comment]
Sorry for jumping the gun, @tarsius 😄 I was just a little worried about letting it sit; I've never used bountysource before this week. I forwarded it to [Magit](https://www.bountysource.com/teams/magit); turn it into `transient.el`! ❤️

Thanks for making plugins like this possible 😄

**Comment by tarsius** [comment]
Well then, thanks to both of you!  ;-)

**Comment by fommil** [comment]
this plugin is basically useless for @tarsius because he doesn't have any issues to browse :stuck_out_tongue:

**Comment by tarsius** [comment]
Did you check lately? I opened about fifty over the last few weeks myself!

**Comment by junjizhi** [comment]
I'm reading this thread and a bit lost. So how do we set magithub's current repo to be other than `origin`? You mentioned magithub proxy in one comment, but that option is not available in the build 20181024.2217.

We have a similar workflow where we have an upstream team repo and each team member has a fork. But we need to create an issue on the team repo before we can submit a pull request from our fork. With magithub I have to rename the remote `upstream` to `origin` to enable submitting an issue. Is there another way? 

Let me know if I should create a separate issue or link to an existing one. Thanks.

**Comment by vermiculus** [comment]
You should be able to set the remote with `H C x` from the status buffer. See also [the manual's section on proxies](https://github.com/vermiculus/magithub/blob/master/magithub.org#proxies).
